processors/src/main/java/io/opentelemetry/contrib/interceptor/common/ComposableInterceptor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.interceptor.common;
+
+import io.opentelemetry.contrib.interceptor.api.Interceptor;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Allows to run an item through a list of interceptors in the order they were added. */
+public final class ComposableInterceptor<T> implements Interceptor<T> {
+  private final List<Interceptor<T>> interceptors = new ArrayList<>();
+
+  public void add(Interceptor<T> interceptor) {
+    if (!interceptors.contains(interceptor)) {
+      interceptors.add(interceptor);
+    }
```

Comment:
Assuming `interceptors` is an instance of `CopyOnWriteArrayList`:
```suggestion
    interceptors.addIfAbsent(interceptor);
```